plugin/pkg/admission/resourcequota/admission.go

```diff
@@ -182,6 +182,12 @@ func (q *quotaAdmission) Admit(a admission.Attributes) (err error) {
 		return nil
 	}
 
+	if om, err := api.ObjectMetaFor(inputObject); namespace != "" && err == nil {
```

Comment:
can you add a comment in the code that explains why this is needed?

the use case as discussed is the following:

a namespace scoped resource may reference non-namespaced resources
the quota is measuring the number of unique references in the namespace
in order to calculate the uniqueness constraint, usage for one object is only knowable by knowing all other object's usage.  to know the other object's usage, we need to know the namespace.